paper-api/src/main/java/org/bukkit/entity/HumanEntity.java

```diff
@@ -428,6 +429,31 @@ default void closeInventory() {
      */
     public void setCooldown(ItemStack item, int ticks);
 
+    /**
+     * Get the cooldown time in ticks remaining for the specified cooldown group.
+     *
+     * @param key the cooldown group to check
+     * @return the remaining cooldown time in ticks
+     * @see UseCooldown#cooldownGroup()
+     */
+    public int getCooldown(NamespacedKey key);
+
+    /**
+     * Set a cooldown on items with the specified cooldown group for a certain amount of ticks.
+     * 0 ticks will result in the removal of the cooldown.
+     * <p>
+     * Cooldowns are used by the server for items such as ender pearls and
+     * shields to prevent them from being used repeatedly.
+     * <p>
+     * Note that cooldowns will not by themselves stop an item from being used
+     * for attacking.
+     *
+     * @param key cooldown group to set the cooldown for
+     * @param ticks the amount of ticks to set or 0 to remove
+     * @see UseCooldown#cooldownGroup()
+     */
+    public void setCooldown(NamespacedKey key, int ticks);
```

Comment:
Can this be an adventure key instead? (the conversation is then in PaperAdventure instead of CraftNamespacedKey)